grid2op/tests/test_alert_trust_score.py

```diff
@@ -0,0 +1,1892 @@
+
+                        assert cm_reward_min_ep == 0.0
+                        assert cm_reward_max_ep == 0.0
+                else:
+                    raise Grid2OpException('No attack expected')
+
+                if done:
+                    break
+
+            assert done
+
+    # return 0
+    def test_assistant_trust_score_blackout_no_attack_before_window_no_alert(self) -> None:
+        """Even if there is a blackout, an we raise an alert too late
+           we expect a score of 0 because there is no attack and we don't finish the scenario"""
+        with make(
+                self.env_nm,
+                test=True,
+                difficulty="1",
+                reward_class=_AlertTrustScore(**DEFAULT_PARAMS_TRUSTSCORE)
+        ) as env:
+            env.seed(0)
+            env.reset()
+
+            done = False
+            for i in range(env.max_episode_duration()):
+                act = self.get_dn(env)
+                if i == 3:
+                    act = self.get_blackout(env)
+                elif i == 4:
+                    # we never go here ...
+                    act = env.action_space({"raise_alert": [0]})
+                obs, score, done, info = env.step(act)
+
+                if info["opponent_attack_line"] is None:
+                    assert score == 0.
+                    if done:
+                        total_nb_attacks = env._reward_helper.template_reward.total_nb_attacks
+                        nb_last_attacks = env._reward_helper.template_reward.nb_last_attacks
+
+                        assert nb_last_attacks == 0
+                        assert total_nb_attacks == 0
+
+                        assert env._reward_helper.template_reward.total_nb_attacks == 0.
+                        assert env._reward_helper.template_reward.cumulated_reward == 0.
+
+                        cm_reward_min_ep, cm_reward_max_ep = env._reward_helper.template_reward._compute_min_max_reward(
+                            total_nb_attacks, nb_last_attacks)
+
+                        assert cm_reward_min_ep == 0.0
+                        assert cm_reward_max_ep == 0.0
+                else:
+                    raise Grid2OpException('No attack expected')
+
+                if done:
+                    break
+
+            assert done
+
+
+class TestAlertTrustScoreBlackout_CauseAttack(unittest.TestCase):
+    """test the basic bahavior of the assistant alert feature when a blackout occur"""
+
+    def setUp(self) -> None:
+        """ WARNING: Parameter ALERT_TIME_WINDOW should be set to 2 in these test for the environment used
+            Max Iter should be set to 10"""
+        self.env_nm = os.path.join(
+            PATH_DATA_TEST, "l2rpn_idf_2023_with_alert"
+        )
+
+    
+    def get_dn(self, env):
+        return env.action_space({})
+
+    def get_blackout(self, env):
+        blackout_action = env.action_space({})
+        blackout_action.gen_set_bus = [(0, -1)]
+        return blackout_action
```

Comment:
same as above, avoid copy paste and prefer reusing the _get_blackout function defined in the _aux_opponent_for_test_blablabla